modules/services/fb_ai_features.js

```diff
@@ -8,37 +8,80 @@ import { osmEntity, osmNode, osmWay } from '../osm';
 import { utilRebind, utilStringQs, utilTiler } from '../util';
 
 // constants
-var API_URL = 'https://mapwith.ai/maps/ml_roads?conflate_with_osm=true&theme=ml_road_vector&collaborator=fbid&token=ASZUVdYpCkd3M6ZrzjXdQzHulqRMnxdlkeBJWEKOeTUoY_Gwm9fuEd2YObLrClgDB_xfavizBsh0oDfTWTF7Zb4C&hash=ASYM8LPNy8k1XoJiI7A&result_type=road_building_vector_xml';
+var APIROOT = 'https://mapwith.ai/maps/ml_roads';
 var TILEZOOM = 16;
 var tiler = utilTiler().zoomExtent([TILEZOOM, TILEZOOM]);
 var dispatch = d3_dispatch('loadedData');
 
+var _datasets = {};
 var _checkpoints = {};
-var _graph;
-var _tileCache;
-var _tree;
 var _deferredAiFeaturesParsing = new Set();
-
 var _off;
 
+
 function abortRequest(i) {
     i.abort();
 }
 
 
-function apiURL(extent, taskExtent) {
+function tileURL(dataset, extent, taskExtent) {
+    // Conflated datasets have a different ID, so they get stored in their own graph/tree
+    var isConflated = /-conflated$/.test(dataset.id);
+    var datasetID = dataset.id.replace('-conflated', '');
+
+    var qs = {
+        conflate_with_osm: isConflated,
+        theme: 'ml_road_vector',
+        collaborator: 'fbid',
+        token: 'ASZUVdYpCkd3M6ZrzjXdQzHulqRMnxdlkeBJWEKOeTUoY_Gwm9fuEd2YObLrClgDB_xfavizBsh0oDfTWTF7Zb4C',
+        hash: 'ASYM8LPNy8k1XoJiI7A'
+    };
+
+    if (datasetID === 'fbRoads') {
+        qs.result_type = 'road_vector_xml';
+
+    } else if (datasetID === 'msBuildings') {
+        qs.result_type = 'road_building_vector_xml';
+        qs.building_source = 'microsoft';
+
+    } else {
+        qs.result_type = 'osm_xml';
+        qs.sources = `esri_building.${datasetID}`;
+    }
+
     // fb_ml_road_url: if set, get road data from this url
-    var fb_ml_road_url = utilStringQs(window.location.hash).fb_ml_road_url;
-    var result = (fb_ml_road_url ? fb_ml_road_url : API_URL) + '&bbox=' + extent.toParam();
-    if (taskExtent) result += '&crop_bbox=' + taskExtent.toParam();
-
-    var custom_ml_road_tags = utilStringQs(window.location.hash).fb_ml_road_tags;
-    if (custom_ml_road_tags) {
-      custom_ml_road_tags.split(',').forEach(function (tag) {
-        result += '&allow_tags[]=' + tag;
-      });
+    // var fb_ml_road_url = utilStringQs(window.location.hash).fb_ml_road_url;
+
+    qs.bbox = extent.toParam();
+
+    // var result = (fb_ml_road_url ? fb_ml_road_url : API_URL) + '&bbox=' + ;
+    if (taskExtent) qs.crop_bbox = taskExtent.toParam();
+
+    // TODO: FIX
```

Comment:
Should we chat about this TODO? I'm not sure this even gets used anymore. Probably another candidate for internal/power user features.

Reply:
I'll leave this one commented for now - I agree it's probably not used.